test/auditing.jl

```diff
@@ -35,37 +90,81 @@ end
                 raw"""
                 cd ${WORKSPACE}/srcdir/isa_tests
-                make -j${nproc} install
+                make -j${nproc} CFLAGS="-march=sandybridge -mtune=sandybridge" install
                 install_license /usr/include/ltdl.h
                 """,
                 # Build for our platform
                 [platform],
                 # Ensure our executable products are built
-                products,
+                [product],
                 # No dependencies
                 Dependency[];
-                # We need to build with very recent GCC so that we can emit AVX2
-                preferred_gcc_version=v"8",
+                preferred_gcc_version=v"6",
+                lock_microarchitecture=false,
             )
+
+            # Extract our platform's build
+            @test haskey(build_output_meta, platform)
+            tarball_path, tarball_hash = build_output_meta[platform][1:2]
+            @test isfile(tarball_path)
+
+            # Unpack it somewhere else
+            @test verify(tarball_path, tarball_hash)
+            testdir = joinpath(build_path, "testdir")
+            mkdir(testdir)
+            unpack(tarball_path, testdir)
+            prefix = Prefix(testdir)
+
+            # Run ISA test
+            readmeta(locate(product, prefix)) do oh
+                detected_isa = Auditor.analyze_instruction_set(oh, platform; verbose=true)
+                @test detected_isa == :sandybridge
+            end
         end
+    end
 
-        # Extract our platform's build
-        @test haskey(build_output_meta, platform)
-        tarball_path, tarball_hash = build_output_meta[platform][1:2]
-        @test isfile(tarball_path)
+    # The microarchitecture of the product matches the target architecture: no warnings!
+    for (march, isa) in (("x86_64", "x86_64"), ("avx", "sandybridge"), ("avx2", "haswell"), ("avx512", "skylake_avx512"))
+        mktempdir() do build_path
+            platform = ExtendedPlatform(Linux(:x86_64); march=march)
+            build_output_meta = nothing
+            @test_logs (:info, "Building for x86_64-linux-gnu-march+$(march)") match_mode=:any begin
+                build_output_meta = autobuild(
+                    build_path,
+                    "isa_tests",
+                    v"1.0.0",
+                    [DirectorySource(build_tests_dir)],
+                    # Build the test suite, install the binaries into our prefix's `bin`
+                    raw"""
+                    cd ${WORKSPACE}/srcdir/isa_tests
+                    make -j${nproc} install
+                    install_license /usr/include/ltdl.h
+                    """,
+                    # Build for our platform
+                    [platform],
+                    # Ensure our executable products are built
+                    [product],
+                    # No dependencies
+                    Dependency[];
+                )
+            end
```

Comment:
Can you add a test where `lock_microarchitecture == true` but you set a `-march` flag that causes the compiler wrapper to complain?

Reply:
This is a test I wanted to have in `BinaryBuilderBase`, I'll add it later, thanks for reminding me!